src/core/server/http/auth_headers_storage.ts

```diff
@@ -19,8 +19,8 @@ export type GetAuthHeaders = (request: KibanaRequest) => AuthHeaders | undefined
 
 /** @internal */
 export class AuthHeadersStorage {
-  private authHeadersCache = new WeakMap<LegacyRequest, AuthHeaders>();
-  public set = (request: KibanaRequest | LegacyRequest, headers: AuthHeaders) => {
+  private authHeadersCache = new WeakMap<Request, AuthHeaders>();
+  public set = (request: KibanaRequest | Request, headers: AuthHeaders) => {
```

Comment:
```suggestion
  public set = (request: KibanaRequest, headers: AuthHeaders) => {
```

Reply:
I misunderstood the suggestion for replacing `LegacyRequest` with `Hapi.Request` as removing the type alias. Thanks for clarifying!

Reply:
We still need to allow `Hapi.Request` as the request param type for the [auth scheme](https://github.com/elastic/kibana/blob/main/src/core/server/http/http_server.ts#L436-L455). If we did want to revisit that, we'd need to do some more refactoring.